src/CBT.NuGet.AggregatePackage.UnitTests/AggregateTest.cs

```diff
@@ -0,0 +1,173 @@
+﻿using Microsoft.Build.Utilities;
+using Microsoft.MSBuildProjectBuilder;
+using Shouldly;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Reflection;
+using System.Xml;
+using Xunit;
+
+namespace CBT.NuGet.AggregatePackage.UnitTests
+{
+    public class AggregateTest : IDisposable
+    {
+        private string _testRootFolder = string.Empty;
+        private string _enlistmentRoot = string.Empty;
+        private string _repo = string.Empty;
+        private string _testProject = string.Empty;
+        private string _aggregatePropsInNupkg = string.Empty;
+        private string _destPackagesPath = string.Empty;
+        private IDictionary<string, string> _installedNupkgs = null;
+        public AggregateTest()
+        {
+            _testRootFolder = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            _enlistmentRoot = Path.Combine(_testRootFolder, "enl1");
+            _repo = Path.Combine(Directory.GetCurrentDirectory(), "repo");
+            _destPackagesPath = Path.Combine(_enlistmentRoot, "packages");
+        }
+
+        public void Dispose()
+        {
+            Directory.Delete(_testRootFolder, true);
+        }
+
+        private IDictionary<string, string> InstallNupkgFiles(string repo, string destinationRoot, string filePattern = "*.nupkg")
+        {
+            IDictionary<string, string> pkgInstalls = new Dictionary<string, string>();
+            foreach (var sourceNupkg in Directory.EnumerateFiles(repo, filePattern))
+            {
+                var destPath = Path.Combine(destinationRoot, Path.GetFileNameWithoutExtension(sourceNupkg));
+                ZipFile.ExtractToDirectory(sourceNupkg, destPath);
+                var nuspec = Directory.EnumerateFiles(destPath, "*.nuspec", SearchOption.TopDirectoryOnly).SingleOrDefault();
+                var packageID = Path.GetFileNameWithoutExtension(nuspec);
+
+                pkgInstalls.Add(packageID, destPath);
+            }
+            return pkgInstalls;
+        }
+
+        private void CreateTestProject(string testProject)
+        {
+            Directory.CreateDirectory(Path.GetDirectoryName(testProject));
+            var testProjectElement = ProjectBuilder.Create()
+                    .AddProperty("CBTEnablePackageRestore=true")
+                    .AddProperty("Platform=AnyCPU", "Configuration=Debug", "OutputPath=bin")
+                    .AddProperty($"CBTNuGetTasksAssemblyPath={Assembly.GetExecutingAssembly().Location}")
+                    .AddProperty("CBTAggregatePackage=NuGetPath_Microsoft_Build=src1|scr2")
+                    .AddProperty($"NuGetPackagesPath={_destPackagesPath}")
+                    .AddProperty($"IntermediateOutputPath={_enlistmentRoot}\\obj\\$(Configuration)\\$(Platform)\\AggregatePackage.proj")
+                    .AddImport($"{_installedNupkgs["CBT.NuGet.AggregatePackage"]}\\cbt\\module\\After.CBT.Nuget.props")
```

Comment:
I always use the `@` symbol to avoid double backslashes, what do you think?  I find it easier to read...